src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -410,80 +705,133 @@ const internalGetFreshAddress = async (
   })
 }
 
-interface ProcessAddressArgs extends FormatArgs {
-  address: string
-}
+interface ProcessFormatAddressesArgs extends FormatArgs {}
 
-const processAddress = async (args: ProcessAddressArgs): Promise<void> => {
-  const { address, blockBook, addressesToWatch, onAddressChecked } = args
-
-  const firstProcess = !addressesToWatch.has(address)
-  if (firstProcess) {
-    addressesToWatch.add(address)
-    blockBook.watchAddresses(
-      Array.from(addressesToWatch),
-      async (response): Promise<void> => {
-        await setLookAhead(args)
-        await processAddress({ ...args, address: response.address })
-      }
-    )
+const processFormatAddresses = async (
```

Comment:
Turn this into generic function that receives args and executor, then use that every time we need to loop over all branches for a format.

Reply:
This is changed again after rebase, so I don't think it's worth touching it again before rebase.